inferred-spans/src/main/java/io/opentelemetry/contrib/inferredspans/internal/SamplingProfiler.java

```diff
@@ -739,18 +777,25 @@ public void start() {
 
   @SuppressWarnings({"FutureReturnValueIgnored", "Interruption"})
   public void reschedule() {
-    Future<?> future = this.profilingTask;
-    if (future != null) {
-      if (future.cancel(true)) {
+    profilerLock.lock();
+    try {
+      Future<?> future = this.profilingTask;
+      if (future != null && future.cancel(true)) {
         Duration profilingDuration = config.getProfilingDuration();
         long delay = config.getProfilingInterval().toMillis() - profilingDuration.toMillis();
         profilingTask = scheduler.schedule(this, delay, TimeUnit.MILLISECONDS);
       }
+    } finally {
+      profilerLock.unlock();
     }
   }
 
+  @SuppressWarnings({"FutureReturnValueIgnored", "Interruption"})
   public void stop() throws InterruptedException, IOException {
     // cancels/interrupts the profiling thread
+    if (profilingTask != null) {
+      profilingTask.cancel(true);
+    }
```

Comment:
[minor] I would suggest to also wrap the access to `profilingTask` and potentially modifying its state with the `profilerLock`, at least for consistency with the other call to `.cancel(true)` above. There is however no side effect on potentially calling this twice as the return value of `cancel(...)` is ignored.